Whole-game.Rmd

```diff
@@ -537,208 +648,327 @@ cat(pretty_install(), sep = "\n")
+You can provide the file's basename or, if you are editing the relevant source file in RStudio, it will be automatically generated.
+For many of you, if `R/strsplit1.R` is the active file in RStudio, you can just call `use_test()`.
+However, since this book is built non-interactively, we must provide the basename explicitly:
 
-```{r test-fbind, eval = create}
-use_test("fbind")
+```{r test-strsplit1, eval = create}
+use_test("strsplit1")
 ```
 
-This creates the file `tests/testthat/test-fbind.R`. Put this content in it:
+This creates the file `tests/testthat/test-strsplit1.R`.
+If it had already existed, `use_test()` would have just opened it.
+Put this content in:
 
-:::downlit
-```{r test-fbind-fodder, echo = FALSE, comment = NA, eval = create}
-test_fodder <- c(
-  'test_that("fbind() binds factor (or character)", {',
-  '  x <- c("a", "b")',
-  '  x_fact <- factor(x)',
-  '  y <- c("c", "d")',
-  '  z <- factor(c("a", "b", "c", "d"))',
-  '',
-  '  expect_identical(fbind(x, y), z)',
-  '  expect_identical(fbind(x_fact, y), z)',
-  '})'
+```{r test-strsplit1-fodder, echo = FALSE, comment = NA, eval = create}
+test_fodder <- glue('
+  test_that("strsplit1() splits a string", {
+    expect_equal(strsplit1("a,b,c", split = ","), c("a", "b", "c"))
+  })',
+  .open = "<<", .close = ">>"
 )
-test_path <- path("tests", "testthat", "test-fbind.R")
+test_path <- path("tests", "testthat", "test-strsplit1.R")
 writeLines(test_fodder, test_path)
 cat(test_fodder, sep = "\n")
 ```
-:::
 
-This tests that `fbind()` gives an expected result when combining two factors and a character vector and a factor.
+This tests that `strsplit1()` gives the expected result when splitting a string.
 
-```{r commit-fbind-test, echo = debug, comment = NA, eval = FALSE}
-git_add(path("tests", "testthat", "test-fbind.R"))
-git_commit("Test fbind()")
+```{r commit-strsplit1-test, echo = debug, comment = NA, eval = FALSE}
+git_add(test_path)
+git_commit("Test strsplit1()")
 ```
 
-Run this test interactively, as you will when you write your own. Note you'll have to attach testthat via `library(testthat)` in your R session first and you'll probably want to `load_all()`.
+Run this test interactively, as you will when you write your own.
+Note you'll have to attach testthat via `library(testthat)` in your R session first and you'll probably want to `load_all()`.
 
 Going forward, your tests will mostly run *en masse* and at arm's length via `test()`:
 
-*TODO: work on the aesthetics of this output. Or maybe testthat 3e will save me the trouble.*
+<!-- TODO: work on the aesthetics of this output. Or maybe testthat 3e will save me the trouble. -->
 
-```{r test, eval = create, message = FALSE, R.options = list(testthat.default_reporter = testthat::ProgressReporter$new(update_interval = Inf))}
+```{r test}
+#| eval = create, message = FALSE,
+#| R.options = list(testthat.default_reporter = testthat::ProgressReporter$new(update_interval = Inf))
```

Comment:
This is not achieving what we want. Currently this chunk shows absolutely nothing.

Reply:
What it should show is more like:

``` r
test()
#> ℹ Loading regexcite
#> ℹ Testing regexcite
#> ✓ | F W S  OK | Context
#> ⠏ |         0 | str_split_one                                                      ✓ |         4 | str_split_one
#> 
#> ══ Results ════════════════════════════════════════════════════════════════════════
#> [ FAIL 0 | WARN 0 | SKIP 0 | PASS 4 ]
#> 
#> 😀
```